### DISCUSSION TITLE
API Reference Documentation?

### DISCUSSION THREAD

**Discussion by sameleff** [ghostfolio/ghostfolio#2605] [Q&A]
Are there reference documents included with the Ghostfolio API, either as a subfolder of the api via something like `/api/v1/docs` or on a standalone Wiki page elsewhere? The [README.md](https://github.com/ghostfolio/ghostfolio/blob/main/README.md) contains a description for the `import` function, but I can't seem to find documentation on the others.

**Comment by dtslvr** [comment]
Hi @sameleff

The public API documented in our [README.md](https://github.com/ghostfolio/ghostfolio#public-api) is considered stable and thus safe to use. However, all internal API endpoints are experimental and subject to change, so you may use them at your own discretion and risk.

**Reply by sameleff** [reply]
I assume that means there is no documentation location else where for the other API endpoints? It looks like all of the code for these is stored under ghostfolio/apps/api/src/app...sounds like reading through the code is the best way to understand how to interact with them?

**Reply by digitlength** [reply]
Hi @sameleff : did you persevere with this? I'm also interested in more API features than just `import`

**Reply by sameleff** [reply]
Hey @digitlength, no, I stopped trying to run my own Ghostfolio service due to an [issue ](https://github.com/ghostfolio/ghostfolio/issues/2640#issuecomment-1984990439) that prevented it from ever running successfully to begin with. That issue has now been resolved, so I'll likely dig into itagain, but unfortunately no progress on this part yet.

**Comment by klausmcm** [comment]
A documented and stable api would be really nice to have. I would love to be able to make an api call to get the transactions from ghostfolio (returned data could be similar to what is returned with the **Export Activities** button under **Portfolio** -> **Activities**).
